src/coreComponents/python/CMakeLists.txt

```diff
@@ -1,61 +1,55 @@
-# virtual environment
-if( NOT PYTHON_POST_EXECUTABLE )
+if ( Python3_EXECUTABLE )
+     # Select the version of python to target
      if( ENABLE_PYGEOSX )
           set( PYTHON_POST_EXECUTABLE ${CMAKE_BINARY_DIR}/lib/PYGEOSX/bin/python CACHE PATH "" FORCE )
      else()
-          set( PYTHON_POST_EXECUTABLE ${PYTHON_EXECUTABLE} CACHE PATH "" FORCE )
+          set( PYTHON_POST_EXECUTABLE ${Python3_EXECUTABLE} CACHE PATH "" FORCE )
      endif()
-endif()
-
-
-blt_import_library( NAME python_interp
-                    INCLUDES    ${PYTHON_INTERP_INCLUDE_DIRS}
-                    HEADERS     ${PYTHON_INTERP_INCLUDE_DIRS}
-                    HEADERS_OUTPUT_SUBDIR python_interp
-                    LIBRARIES python2.7 
-                  )
-
 
-set( GEOSX_XML_TOOLS_BINS
-     "${CMAKE_BINARY_DIR}/bin/preprocess_xml"
-     "${CMAKE_BINARY_DIR}/bin/format_xml"
-   )
+     # Check for the virtualenv package
+     execute_process(
+         COMMAND ${Python3_EXECUTABLE} -c "import virtualenv"
+         RESULT_VARIABLE VIRTUALENV_AVAILABLE
+     )
 
+     if (NOT ${VIRTUALENV_AVAILABLE} EQUAL 0)
+         message(WARNING "To build the PYGEOSX and/or the GEOSX python tools, \"virtualenv\" should be installed in the target python.")
+         message(STATUS "To install virtualenv, try running ${PYTHON_POST_EXECUTABLE} -m pip install virtualenv")
+     endif()
 
-add_custom_command( OUTPUT ${GEOSX_XML_TOOLS_BINS}
-                    COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/setupPythonEnvironment.bash -p ${PYTHON_POST_EXECUTABLE} -b ${CMAKE_BINARY_DIR}/bin
-                    WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-                  )
-
-add_custom_target( geosx_xml_tools 
-                   DEPENDS ${GEOSX_XML_TOOLS_BINS} )
-
-add_custom_target( geosx_xml_tools_test
-                   COMMAND ${CMAKE_BINARY_DIR}/python/geosx/bin/test_geosx_xml_tools
-                   COMMAND rm -r ${CMAKE_BINARY_DIR}/python/geosx_xml_tools_tests*
-                   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/python
-                   DEPENDS geosx_xml_tools 
-                 )
-
-add_custom_target( geosx_format_all_xml_files
-                   COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/formatXMLFiles.bash -g ${CMAKE_BINARY_DIR}/bin/format_xml ${CMAKE_SOURCE_DIR} ${CMAKE_SOURCE_DIR}/../examples
-                   WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
-                   DEPENDS geosx_xml_tools 
-                 )
+     # Build targets
+     set( GEOSX_PYTHON_TOOLS_BINS
+          "${CMAKE_BINARY_DIR}/bin/preprocess_xml"
+          "${CMAKE_BINARY_DIR}/bin/format_xml"
+        )
+
+     add_custom_command( OUTPUT ${GEOSX_PYTHON_TOOLS_BINS}
+                         COMMAND bash ${CMAKE_SOURCE_DIR}/../scripts/setupPythonEnvironment.bash -p ${PYTHON_POST_EXECUTABLE} -b ${CMAKE_BINARY_DIR}/bin
+                         WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+                       )
+
+     add_custom_target( geosx_python_tools 
```

Comment:
since we are adding a new target we could call it directly `geos_python_tools`. but it's also fine if we want to change all targets names at once